internal/core/docker_host.go

```diff
@@ -262,6 +266,16 @@ func dockerHostFromContext(ctx context.Context) (string, error) {
 	return "", ErrDockerSocketNotSetInContext
 }
 
+// dockerHostFromContext returns the docker host from the Go context, if it's not empty
+func dockerHostFromDockerContext(ctx context.Context) (string, error) {
+	dockerHost, err := GetDockerHostFromCurrentContext()
+	if err == nil {
+		return dockerHost, nil
+	}
+
+	return "", ErrDockerSocketNotSetInDockerContext
```

Comment:
question: Is this discarding useful information from the actual error?

Reply:
Indeed. Wdyt about this order: return "", errors.Join(ErrDockerSocketNotSetInDockerContext, err)?
```suggestion
	return "", errors.Join(ErrDockerSocketNotSetInDockerContext, err)
```

Reply:
I think that would be confusing, I would detect a specific "not found" error from the call and only translate that if needed. If that's not needed I would just return a wrapped straight wrapped error.

Reply:
Wdyt https://github.com/testcontainers/testcontainers-go/pull/2810/commits/a38a34591248c2d0205eb1a6ef60ef1686f25520 ?